onmt/translate/beam_search.py

```diff
@@ -189,18 +189,18 @@ def update_finished(self):
                     self.is_finished[i].all()
             else:
                 finish_flag = self.top_beam_finished[i] != 0
-            if finish_flag and len(self.hypotheses[b]) >= self.n_best:
+            if finish_flag and len(self.hypotheses[b]) >= self.beam_size:
                 best_hyp = sorted(
-                    self.hypotheses[b], key=lambda x: x[0], reverse=True)
+                    self.hypotheses[b], key=lambda x: x[0],
```

Comment:
Could everything after the if statement be moved in a method with a clear name (register_hypotheses for ex) ?

Reply:
in another PR please ....